### ISSUE TITLE
Add key-based lookup of Tests in TestSuites

### ISSUE THREAD

**Issue by rgerkin** [scidash/sciunit#110]
We would like to be able to do this:

```
from sciunit import Test, TestSuite
empty_observation = {}
test1 = Test(empty_observation, name='smith')
test2 = Test(empty_observation, name='jones')
suite = TestSuite([test1, test2])
x = suite['smith']
x == test1, x == test2
```

And have it print `True, False` since `x` refers to `test1` (with name 'smith') and not to `test2`. So this will be basically equivalent to the TestSuite being a dictionary, as long as the keys you want to use are set as the test names.

**Comment by rgerkin** [comment]
Implemented in deb589dd706567a26481de1b9928a982815ff092.

**Comment by russelljjarvis** [comment]
Good to see. I think this will improve productivity in downstream code.
